src/pelib/RichHeader.cpp

```diff
@@ -54,6 +734,62 @@ namespace
 		validStructure = (decryptedHeader.size() >= 4);
 	}
 
+	void RichHeader::getUserFriendlyProductName(PELIB_IMAGE_RICH_HEADER_RECORD & record)
+	{
+		// Product ID can be mapped to Product name 1:1. Just check if the ID is in range.
+		record.ProductName = (record.ProductId < productNames.size()) ? productNames[record.ProductId] : "Unknown";
+
+		// Estimate the Visual Studio version
+		if (!(record.ProductId >= 0x83 && record.ProductBuild == 50727))
```

Comment:
Do these values have some kind of meaning you can assign to them? I would personally prefer if these were hidden behind some constant with a name if possible.

Reply:
I'll explain by comment. Defined constant won't work in this case.